radixdlt-java/src/main/java/com/radixdlt/client/application/translate/data/SendMessageToParticlesMapper.java

```diff
@@ -15,52 +17,79 @@
 import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.function.Function;
 import java.util.function.Supplier;
+import java.util.stream.Stream;
 
 /**
  * Maps a send message action to the particles necessary to be included in an atom.
  */
 public class SendMessageToParticlesMapper implements ActionToParticlesMapper {
+
+	/**
+	 * A module capable of creating new securely random ECKeyPairs
+	 */
 	private final Supplier<ECKeyPair> generator;
 
+	/**
+	 * Function which runs every time a mapping is requested via map(action).
+	 * Determines what public keys to encrypt a message with given an application level
+	 * SendMessageAction.
+	 */
+	private final Function<SendMessageAction, Stream<ECPublicKey>> encryptionScheme;
+
+	/**
+	 * New SendMessage action mapper which by default adds both sender and receiver
+	 * public keys as readers of encrypted messages
+	 *
+	 * @param generator module to be used for creating new securely random ECKeyPairs
+	 */
 	public SendMessageToParticlesMapper(Supplier<ECKeyPair> generator) {
+		this(generator, sendMsg -> Stream.of(sendMsg.getFrom(), sendMsg.getTo()).map(RadixAddress::getPublicKey));
+	}
+
+	/**
+	 * SendMessage action mapper which uses a given eckeypair generator and encryption
+	 * scheme
+	 *
+	 * @param generator module to be used for creating new securely random ECKeyPairs
+	 * @param encryptionScheme function to decide which public keys to encrypt wiht
+	 */
+	public SendMessageToParticlesMapper(Supplier<ECKeyPair> generator, Function<SendMessageAction, Stream<ECPublicKey>> encryptionScheme) {
 		this.generator = generator;
+		this.encryptionScheme = encryptionScheme;
 	}
 
+	/**
+	 * If SendMessageAction is unencrypted, returns a single message particle containing the
+	 * payload data.
+	 *
+	 * If SendMessageAction is encrypted, creates a private key encrypted by both from and to
+	 * users, stores that into a message particles and then creates another message particle
+	 * with the payload encrypted by the newly created private key.
+	 *
+	 * @param action the action to map to particles
+	 * @return observable of spunparticles to be included in an atom for a given action
+	 */
 	public Observable<SpunParticle> map(Action action) {
```

Comment:
Realise this hasn't changed - why is this an `Observable` instead of, say, a `List` or `Collection`. Doesn't seem like there is anything async going on there to me.

Reply:
Missing an @Override to show that this method is part of interface.